ddtrace/tracer/tracer.go

```diff
@@ -143,24 +170,51 @@ func newTracer(opts ...StartOption) *tracer {
 		prioritySampling: newPrioritySampler(),
 		pid:              strconv.Itoa(os.Getpid()),
 	}
+	t.config.statsd.Incr("datadog.tracer.started", nil, 1)
 	if c.runtimeMetrics {
-		statsd, err := statsd.NewBuffered(t.config.dogstatsdAddr, 40)
-		if err != nil {
-			log.Warn("Runtime metrics disabled: %v", err)
-		} else {
-			log.Debug("Runtime metrics enabled.")
-			go t.reportMetrics(statsd, defaultMetricsReportInterval)
-		}
+		log.Debug("Runtime metrics enabled.")
+		go t.reportMetrics(defaultMetricsReportInterval)
 	}
-
+	t.wg.Add(2)
 	go t.worker()
-
+	go t.statsSender()
 	return t
 }
 
+func (t *tracer) sendSpanStats() {
+	spansStarted := atomic.SwapInt64(&t.spansStarted, 0)
+	spansFinished := atomic.SwapInt64(&t.spansFinished, 0)
+	tracesDropped := atomic.SwapInt64(&t.tracesDropped, 0)
+	spansDropped := atomic.SwapInt64(&t.spansDropped, 0)
+	t.config.statsd.Count("datadog.tracer.spans.started", spansStarted, nil, 1)
+	t.config.statsd.Count("datadog.tracer.spans.finished", spansFinished, nil, 1)
+	t.config.statsd.Count("datadog.tracer.traces.dropped", tracesDropped, nil, 1)
+	t.config.statsd.Count("datadog.tracer.spans.dropped", spansDropped, nil, 1)
+}
+
+func (t *tracer) statsSender() {
+	defer t.wg.Done()
+	ticker := time.NewTicker(statsInterval)
+	defer ticker.Stop()
+	defer func() {
+		t.config.statsd.Close()
```

Comment:
You could just put this under the t.stopped case then return?

Reply:
👍